### DISCUSSION TITLE
Hasura Claims Map?

### DISCUSSION THREAD

**Discussion by cj** [nhost/nhost#815] [Q&A]
Hi,

Is it possible to use the Hasura Claims Map https://hasura.io/docs/latest/graphql/core/auth/authentication/jwt/#claims_map

It looks to be hard coded here https://github.com/nhost/nhost/blob/2a684d3f8474a100fc5a0d442a7f908102cfa272/packages/core/src/types.ts#L154-L172

Cheers!

**Comment by elitan** [comment]
I don't think we have support for it, no.

What is your use case? What prompted you to ask?

**Reply by cj** [reply]
@elitan for security reasons we need to obfuscate our stack, having x-hasura gives that away 😄 I think for Enterprise customs that would be a big thing.

**Reply by cj** [reply]
@elitan I just wanted to follow up on this.  Would you be open to a pull request?

**Comment by xmlking** [comment]
we are using our own auth provider [authjs](https://authjs.dev)
and using cookies to set token in the browser.  we can make JWT token size small by mapping long `x-hasura-x` fields using `claims_map` so we don't exceed the cookie limit.

example: 
```json
    {
          "type": "HS256",
          "key": "11111",
          "claims_map": {
            "x-hasura-allowed-roles": { "path": "$.roles" },
            "x-hasura-default-role": "anonymous",
            "x-hasura-user-id": { "path": "$.email" },
            "x-hasura-org-id":  { "path": "$.org", "default": "public" }
          }
        }
```
